src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -4442,17 +4469,18 @@ InvocationOperation makeInvocationDroppingInstanceForStaticMethods(IMethodSymbol
             InvocationOperation makeInvocation(SyntaxNode syntax, IMethodSymbol method, IOperation? instanceOpt, ImmutableArray<IArgumentOperation> arguments)
             {
                 Debug.Assert(method.IsStatic == (instanceOpt == null));
+                var visitedArguments = makeArguments(arguments, ref instanceOpt);
                 return new InvocationOperation(method, instanceOpt,
-                                                isVirtual: method.IsVirtual || method.IsAbstract || method.IsOverride,
-                                                makeArguments(arguments), semanticModel: null, syntax,
-                                                method.ReturnType, isImplicit: true);
+                                               isVirtual: method.IsVirtual || method.IsAbstract || method.IsOverride,
+                                               visitedArguments, semanticModel: null, syntax,
+                                               method.ReturnType, isImplicit: true);
             }
 
-            ImmutableArray<IArgumentOperation> makeArguments(ImmutableArray<IArgumentOperation> arguments)
+            ImmutableArray<IArgumentOperation> makeArguments(ImmutableArray<IArgumentOperation> arguments, ref IOperation? instanceOpt)
             {
                 if (arguments != null)
                 {
-                    return VisitArguments(arguments);
+                    return VisitArray(arguments, UnwrapArgument, RewriteArgumentFromArray);
```

Comment:
>VisitArray [](http://example.com/codeflow?start=27&length=10)

I think that in the long term it would be better to keep calling VisitArguments with appropriate value for the second parameter.